src/System.Diagnostics.TraceSource/tests/TraceClassTests.cs

```diff
@@ -341,9 +351,42 @@ public void TraceTest01()
         }
 
         [Fact]
```

Comment:
right, I looked at  that before and learned from @AlexGhiondea AppContext check will not work pre-4.7. 

I had some other workaround by checking where corelib location but this still a hack. RuntimeInformation would be the best here.